reconcile/utils/terrascript_aws_client.py

```diff
@@ -2573,18 +2573,30 @@ def populate_tf_resource_role(self, spec):
 
         self.add_resources(account, tf_resources)
 
+    def populate_iam_policy(self, account: str, name: str, policy: dict[str, Any]):
+        tf_aws_iam_policy = aws_iam_policy(
+            f"{account}-{name}", name=name, policy=json.dumps(policy)
+        )
+        self.add_resource(account, tf_aws_iam_policy)
+
     def populate_saml_iam_role(
         self,
         account: str,
         name: str,
         saml_provider_name: str,
-        policies: list[str],
+        aws_managed_policies: list[str],
+        customer_managed_policies: list[str] | None = None,
         max_session_duration_hours: int = 1,
     ) -> None:
         """Manage the an IAM role needed for SAML authentication."""
         managed_policy_arns = [
             f"arn:{self._get_partition(account)}:" + f"iam::aws:policy/{policy}"
-            for policy in policies
+            for policy in aws_managed_policies
+        ]
+        managed_policy_arns += [
+            f"arn:{self._get_partition(account)}:"
+            + f"iam::{self.uids[account]}:policy/{policy}"
```

Comment:
no need for `+`, can directly put it in one string, just 2 lines in code if one line is too long.

```python
            f"arn:{self._get_partition(account)}:"
            f"iam::{self.uids[account]}:policy/{policy}"
```

same for the `aws_managed_policies` part.

Reply:
Fixed in [03d472e](https://github.com/app-sre/qontract-reconcile/pull/4184/commits/03d472e3315e211202cea88e10d1ce4b602dc1a2)